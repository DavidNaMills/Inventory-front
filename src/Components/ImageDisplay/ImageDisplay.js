import React from 'react';
import classes from './imagedisplay.module.scss';
import noImage from './noImage.jpg';

const ImageDisplay = (props) =>{
    const{url, alt='', size='medium', blur=false, dispNoImage=true} = props;
    const style = [classes.imageDisplay];

    switch(size){
        case 'small': style.push(classes.imageDisplay__small); break;
        case 'medium': style.push(classes.imageDisplay__medium); break;
        case 'large': style.push(classes.imageDisplay__large); break;
        case 'desktop': style.push(classes.imageDisplay__desktop); break;
        default: style.push(classes.imageDisplay__medium); break;
    }
    
    if(blur){
        style.push(classes.imageDisplay__blur);
    }

    return (
        <img src={url ? url : dispNoImage ? noImage : null} alt={alt} className={style.join(' ')}/>
    )
}

export default React.memo(ImageDisplay);