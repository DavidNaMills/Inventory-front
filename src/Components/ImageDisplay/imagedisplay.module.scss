.imageDisplay{
    padding: '5px';
    margin: '10px';
    margin-bottom: '20px';

    &__blur{
        -webkit-box-shadow: 5px 6px 11px 6px rgba(74,74,74,0.76);
        -moz-box-shadow: 5px 6px 11px 6px rgba(74,74,74,0.76);
        box-shadow: 5px 6px 11px 6px rgba(74,74,74,0.76);
    }

    &__small{
        width: auto;
        max-width: 75px;
    }

    &__medium{
        width: auto;
        max-width: 250px;
    }

    &__large{
        width: auto;
        max-width: 300px;
        height: auto
    }

    &__desktop {
        width: auto;
        max-width: 550px;
        height: auto
    }
}