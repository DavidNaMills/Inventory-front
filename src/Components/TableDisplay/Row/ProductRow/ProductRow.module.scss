.productRow{
    display: flex;
    align-items: center;
    
    &__infoCont{
        display: flex;
        flex-direction: column;
        margin-left: 20px;

    }
    
    

    &__cellCont{
        width: 100%;
        padding-top: 20px;
        display: flex;

        p{
            margin-right: 50px;
        }

        p:last-of-type {
            margin-right: none;
        }

    }

    &__cellCont_stack{
        width: 100%;
        padding-top: 10px;
        padding-bottom: 10px;
        display: flex;
        flex-direction: column;

        p{
            margin-right: 50px;
        }

        p:last-of-type {
            margin-right: none;
        }

    }



    span{
        font-weight: bold;
        margin-right: 10px;
    }
    
}
