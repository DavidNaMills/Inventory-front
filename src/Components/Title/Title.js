import React from 'react';
import classes from './title.module.scss';


const Title = (props) => {
    const { size = null, message = '', center = false } = props;
    let element = null;

    switch (size) {
        case 'h1':
            element = <h1 className={[classes.title_h1, center ? classes.title__center : null].join(' ')} >{message}</h1>
            break;
        case 'h2':
            element = <h2 className={[classes.title_h2, center ? classes.title__center : null].join(' ')}>{message}</h2>
            break;
        case 'h3':
            element = <h3 className={[classes.title_h3, center ? classes.title__center : null].join(' ')}>{message}</h3>
            break;
        case 'h4':
            element = <h4 className={[classes.title_h4, center ? classes.title__center : null].join(' ')}>{message}</h4>
            break;
        case 'h5':
            element = <h5 className={[classes.title_h5, center ? classes.title__center : null].join(' ')}>{message}</h5>
            break;
        case 'h6':
            element = <h6 className={[classes.title_h6, center ? classes.title__center : null].join(' ')}>{message}</h6>
            break;
        default:
            element = <h2 className={[classes.title_h2, center ? classes.title__center : null].join(' ')}>{message}</h2>
            break;
    };

    return (
        <div>
            {element}
        </div>
    )
}

export default React.memo(Title);