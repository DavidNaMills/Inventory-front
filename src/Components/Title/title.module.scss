@import '../../styles/variables/colors.module.scss';
@import '../../styles/fonts/fontSizes.module.scss';
@import '../../styles/mixins/title.module.scss';

.title {
    @include setTitle($h2, $title-color);
    
    &_h1{
        @include setTitle($h1, $title-color);
    };

    &_h2{
        @include setTitle($h2, $title-color);
    };
    
    &_h3{
        @include setTitle($h3, $title-color);
    }

    &_h4{
        @include setTitle($h4, $title-color);
    }

    &_h5{
        @include setTitle($h5, $title-color);
    }

    &_h6{
        @include setTitle($h6, $title-color);
    }

    &__center{
        text-align: center;
    }


}