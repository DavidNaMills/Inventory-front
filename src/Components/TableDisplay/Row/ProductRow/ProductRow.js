import React from 'react';
import { isMobileOnly } from 'react-device-detect';
import classes from './ProductRow.module.scss';
import ImageDisplay from '../../../ImageDisplay/ImageDisplay';
import Title from '../../../Title/Title';
import { useLocale } from '../../../../Context/LocaleContext';

const ProductRow = (props) => {
    const { config, data, withTitle=true, withImage=true, withInfo=true, stack=false } = props;
    const locale = useLocale();

    return (
        <div className={classes.productRow}>
            {withImage && <ImageDisplay url={data.url ? `/productImages/${data.url}` : null} alt={`${data.name}-${data.code}`} size='medium'/>}
            <div className={classes.productRow__infoCont}>
                {withTitle && <Title message={data[config.main[0]]} size='h4' />}
                {withInfo &&
                    <div className={stack? classes.productRow__cellCont_stack : classes.productRow__cellCont}>
                        <p><span>{locale[config.main[1]]}:</span>{data[config.main[1]]}</p>
                        <p><span>{locale[config.main[2]]}:</span>{data[config.main[2]]}</p>
                    </div>
                }
            </div>
        </div>
    )
}

export default ProductRow;