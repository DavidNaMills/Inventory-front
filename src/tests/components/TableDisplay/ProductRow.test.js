import React from 'react';
import '../../testHelpers/context';
import renderer from 'react-test-renderer';
import { mount, shallow } from 'enzyme';
import * as device from 'react-device-detect';

import ProductRow from '../../../Components/TableDisplay/Row/ProductRow/ProductRow';
import ImageDisplay from '../../../Components/ImageDisplay/ImageDisplay';
import Title from '../../../Components/Title/Title';

const baseProps = {
    config: {main : ['name', 'location', 'qty']},
    data: {
        name: 'Blue dogs',
        location: 'Dundee',
        qty: '12345678',
        price: '187.56',
        code: 'abc123',
        url: 'test.jpg'
    }
}

describe('<ProductRow/> test suite', ()=>{
    it('matches snapshot', ()=>{
        const comp = renderer.create(<ProductRow {...baseProps}/>);
        const app = comp.toJSON();
        expect(app).toMatchSnapshot();
    });
    
    it('exists', ()=>{
        const wrapper = shallow(<ProductRow {...baseProps} />);
        expect(wrapper.exists).toBeTruthy();
    });
    
    it('renders a container <div> with className: "productRow"', ()=>{
        const wrapper = shallow(<ProductRow {...baseProps} />);
        const div = wrapper.find('div');
        expect(div.length).toBe(3);
        expect(div.at(0).props()).toHaveProperty('className', 'productRow');
    });
    
    describe('default props values', ()=>{
        const wrapper = mount(<ProductRow {...baseProps} />);

        it('renders an inner <div> with className: "productRow__infoCont"', ()=>{
            expect(wrapper.find('div').at(1).props()).toHaveProperty('className', 'productRow__infoCont');
        });
        
        it('renders an <ImageDisplay/> <Title/> components', ()=>{
            expect(wrapper.find(ImageDisplay).length).toBe(1);
            expect(wrapper.find(ImageDisplay).props()).toHaveProperty('url', '/productImages/test.jpg');
            expect(wrapper.find(ImageDisplay).props()).toHaveProperty('alt', `${baseProps.data.name}-${baseProps.data.code}`);
            expect(wrapper.find(ImageDisplay).props()).toHaveProperty('size', 'medium');
            
            expect(wrapper.find(Title).length).toBe(1);
            expect(wrapper.find(Title).props()).toHaveProperty('message', baseProps.data.name);
            expect(wrapper.find(Title).props()).toHaveProperty('size', 'h4');
        });
        
        it('renders a third <div> with className: "productRow__cellCont"', ()=>{
            expect(wrapper.find('div').at(3).props()).toHaveProperty('className', 'productRow__cellCont');
        });
        
        it('renders 2 <p> elements with data', ()=>{
            const p = wrapper.find('p');
            expect(p.length).toBe(2);
            expect(p.at(0).text()).toEqual(`${baseProps.config.main[1]}:${baseProps.data[baseProps.config.main[1]]}`);
            expect(p.at(1).text()).toEqual(`${baseProps.config.main[2]}:${baseProps.data[baseProps.config.main[2]]}`);
        });
    });

    describe('Optional props tests', ()=>{
        it('does not render an <ImageDisplay/> component', ()=>{
            const tW = shallow(<ProductRow {...baseProps} withImage={false} />);
            expect(tW.find(ImageDisplay).length).toBe(0);
        });
        
        it('does not render a <Title/> component if withTitle is false', ()=>{
            const tW = shallow(<ProductRow {...baseProps} withTitle={false} />);
            expect(tW.find(Title).length).toBe(0);
        });
        
        it('only renders 2 <div> elements with classNames: "productRow" and "productRow__infoCont" when withInfo is false', ()=>{
            const tW = shallow(<ProductRow {...baseProps} withInfo={false} />);
            const p = tW.find('div');
            expect(p.length).toBe(2);
            expect(p.at(0).props()).toHaveProperty('className', 'productRow');
            expect(p.at(1).props()).toHaveProperty('className', 'productRow__infoCont');
        });
    });


});