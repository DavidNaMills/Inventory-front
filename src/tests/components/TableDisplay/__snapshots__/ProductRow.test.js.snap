// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ProductRow/> test suite matches snapshot 1`] = `
<div
  className="productRow"
>
  <img
    alt="Blue dogs-abc123"
    className="imageDisplay imageDisplay__medium"
    src="/productImages/test.jpg"
  />
  <div
    className="productRow__infoCont"
  >
    <div>
      <h4
        className="title_h4 "
      >
        Blue dogs
      </h4>
    </div>
    <div
      className="productRow__cellCont"
    >
      <p>
        <span>
          Locations
          :
        </span>
        Dundee
      </p>
      <p>
        <span>
          :
        </span>
        12345678
      </p>
    </div>
  </div>
</div>
`;
